npc_gzip/utils.py

```
@@ -0,0 +1,90 @@
+import random
+import string
+
+from npc_gzip.exceptions import InvalidObjectTypeException
+
+"""
+Helper functions used primarily in testing the
+rest of the codebase.
+
+>>> number_of_sentences = random.randint(1, 100)
+>>> dataset = generate_dataset(number_of_sentences)
+>>> assert len(dataset) == number_of_sentences
+"""
+
+
+def generate_sentence(number_of_words: int = 10) -> str:
+    """
+    Generates a sentence of random
+    numbers and letters, with 
+    `number_of_words` words in the 
+    sentence such that len(out.split()) \
+    == `number_of_words`.
+
+    Arguments:
+        number_of_words (int): The number of words you
+                               want in the sentence.
+
+    Returns:
+        str: Sentence of random numbers and letters.
+    """
+
+    if not isinstance(number_of_words, int):
+        try:
+            number_of_words = int(number_of_words)
```

I suspect it would be better simply not to support other types than `int` for `number_of_words`, and either raise `InvalidObjectTypeException` whenever it is not of the expected type, or allow `>` or `range` to raise `TypeError`.

If someone passes a value like `5.2`, I think it would be better to get a `TypeError` than to silently convert to `5`. (Values like `5.999999999999999` will also silently convert to `5`, so if the goal is to support using the result of a floating-point calculation with intended rounding, this doesn't do that reliably.)